MediaGalleryUi/Model/UpdateAssetInGrid.php

```diff
@@ -27,110 +24,63 @@ class UpdateAssetInGrid
+    private $logger;
 
     /**
      * Constructor
      *
      * @param ResourceConnection $resource
-     * @param Repository $assetRepository
-     * @param Emulation $emulation
-     * @param ScopeConfigInterface $scopeConfig
+     * @param DriverInterface $file
+     * @param LoggerInterface $logger
      */
     public function __construct(
         ResourceConnection $resource,
-        Repository $assetRepository,
-        Emulation $emulation,
-        ScopeConfigInterface $scopeConfig
+        DriverInterface $file,
+        LoggerInterface $logger
     ) {
         $this->resource = $resource;
-        $this->assetRepository = $assetRepository;
-        $this->appEmulation = $emulation;
-        $this->scopeConfig = $scopeConfig;
+        $this->file = $file;
+        $this->logger = $logger;
     }
 
     /**
-     * Update the grid table for the asset
+     * Update the media asset grid with additional data
      *
      * @param AssetInterface $asset
+     *
      * @return void
+     * @throws CouldNotSaveException
      */
     public function execute(AssetInterface $asset): void
     {
-        $this->getConnection()->insertOnDuplicate(
-            $this->resource->getTableName('media_gallery_asset_grid'),
-            [
-                'id' => $asset->getId(),
-                'directory' => dirname($asset->getPath()),
-                'thumbnail_url' => $asset->getPath(),
-                'preview_url' => $asset->getPath(),
-                'name' => basename($asset->getPath()),
-                'content_type' =>  strtoupper(str_replace("image/", "", $asset->getContentType())),
-                'source_icon_url' => $this->getIconUrl($asset),
-                'width' => $asset->getWidth(),
-                'height' => $asset->getHeight(),
-                'created_at' => $asset->getCreatedAt(),
-                'updated_at' => $asset->getUpdatedAt()
-            ]
-        );
-    }
-
-    /**
-     * Retrieve the database adapter
-     *
-     * @return AdapterInterface
-     */
-    private function getConnection(): AdapterInterface
-    {
-        return $this->resource->getConnection();
-    }
-
-    /**
-     * Return AStock incon url if source is Adobe Stock
-     *
-     * @return ?string
-     */
-    private function getIconUrl(AssetInterface $asset): ?string
-    {
-        $iconUrl = null;
-
-        $this->appEmulation->startEnvironmentEmulation(
-            Store::DEFAULT_STORE_ID,
-            Area::AREA_ADMINHTML,
-            true
-        );
-
-        if (!empty($asset->getSource())) {
-            $iconUrl = $this->assetRepository->getUrlWithParams(
-                'Magento_MediaGalleryUi::images/Astock.png',
-                ['_secure' => $this->getIsSecure()]
+        try {
+            $this->resource->getConnection()->insertOnDuplicate(
+                $this->resource->getTableName('media_gallery_asset_grid'),
+                [
+                    'id' => $asset->getId(),
```

Comment:
Could be better to have keys are constants in resource model or interface

Reply:
Hi @hws47a ,

Thank you for the suggestions. I implemented the majority of them except this one. Can we have a short conversation about this in comments here?

I think that putting constants into interfaces (with the api notation) and resource models (without api notations or with it) makes implementation more coupled and dependent on the source model. Also, as I can see, in this project we commonly do not use constants in interfaces or source models.

Should we change this way @sivaschenko ?

Reply:
Currently we don't have the model so the only place where these fields are declared is db schema. Let's discuss if we need to introduce the model/interface at the next public meeting